packages/eslint-plugin/src/rules/ban-context-export/index.js

```diff
@@ -11,12 +13,9 @@ const minimatch = require('minimatch');
  * }} Options
  */
 
-/** @type {Options} */
-const defaultOptions = {};
-
 module.exports = createRule({
   name: 'ban-context-export',
-  defaultOptions: [],
+  defaultOptions: /** @type {ReadonlyArray<Options>} */ ([{}]),
```

Comment:
with this `createRule` properly instantiates context generic, thus type intellisense works as expected (even in tests)